### PR TITLE
Adding support to use correspondences generated by SuperPoint.

### PR DESCRIPTION
**Pull request by punit-bhatt**
Includes vizualization code changes as well. TODO: CUDA support over multiple workers.

Note: Each CPU iter takes 1.3-1.5 months

### CODE REVIEW

File: correspondences/superpoint_matcher.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+    def __init__(self, opt=None) -> None:
+        super().__init__()
+
+        self.fe = SuperPointFrontend(
+            weights_path=opt.weights_path if opt is not None else 'assets/superpoint_v1.pth',
+            nms_dist=opt.nms_dist if opt is not None else 4,
+            conf_thresh=opt.conf_thresh if opt is not None else 0.015,
+            nn_thresh=opt.nn_thresh if opt is not None else 0.7,
+            cuda=opt.cuda if opt is not None else False,
```
Comment (punit-bhatt) [inline correspondences/superpoint_matcher.py:232]: To just cache the superpoint features - Make the default value True and set the number of data workers to 1. Run training over train and val for 1 epoch each.

File: trainer.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def __init__(self, options):
             self.opt.data_path, train_filenames, self.opt.height, self.opt.width,
             self.opt.frame_ids, 4, is_train=True, img_ext=img_ext,
             load_corresp=self.opt.load_corresp, corresp_cache_path=self.opt.corresp_cache_path,
-            corresp_n=self.opt.corresp_n)
+            corresp_n=self.opt.corresp_n, corresp_type=self.opt.corresp_type)
         self.train_loader = DataLoader(
             train_dataset, self.opt.batch_size, True,
             num_workers=self.opt.num_workers, pin_memory=True, drop_last=True,
```
Comment (punit-bhatt) [inline trainer.py:139]: remove this parameter rather than setting num_workers=1

File: trainer.py [hunk 3]
```diff
@@ -142,7 +142,7 @@ def __init__(self, options):
             self.opt.data_path, val_filenames, self.opt.height, self.opt.width,
             self.opt.frame_ids, 4, is_train=False, img_ext=img_ext,
             load_corresp=self.opt.load_corresp, corresp_cache_path=self.opt.corresp_cache_path,
-            corresp_n=self.opt.corresp_n)
+            corresp_n=self.opt.corresp_n, corresp_type=self.opt.corresp_type)
         self.val_loader = DataLoader(
             val_dataset, self.opt.batch_size, True,
             num_workers=self.opt.num_workers, pin_memory=True, drop_last=True,
```
Comment (punit-bhatt) [inline trainer.py:148]: remove this parameter rather than setting num_workers=1

**Comment by g-meghana-reddy** [comment]
LGTM

**Pull request merged**